### ISSUE TITLE
[Crash] Out of memory while hashing (cannot install big app)

### ISSUE THREAD

**Issue by opk12** [Droid-ify/client#475]
**Describe the bug**
Droid-ify crashes after finishing downloading the latest OsmAnd and before installing it. Note that OsmAnd 4.5.8 is 139 MB ([versions at F-droid](https://f-droid.org/en/packages/net.osmand.plus/)). I closed other apps, background services, etc.

```
FATAL EXCEPTION: main
Process: com.looker.droidify, PID: 27996
java.lang.OutOfMemoryError: Failed to allocate a 145104804 byte allocation with 12582912 free bytes and 115MB until OOM
	at kotlin.text.RegexKt.readBytes(SourceFile:1)
	at com.looker.core.common.signature.HashCheckerKt$readBytesFrom$2.invokeSuspend(SourceFile:26)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile)
	at kotlinx.coroutines.DispatchedTask.run(SourceFile)
	at androidx.work.Worker$2.run(SourceFile:753)
	at kotlinx.coroutines.scheduling.TaskImpl.run(SourceFile)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:94)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@bf25cdb, Dispatchers.Main]
```

**To Reproduce**
1. Download the latest OsmAnd.
2. Wait for the download to finish.
3. When the download is finished, before installing, the app crashes.

**Smartphone (please complete the following information):**
Droid-ify 0.5.8.3 from F-droid, on Samsung Android 7.1

**Additional context**
As a desktop Linux (not Android) programmer: the trace says that the file is copied into the JVM's heap for hashing. Is there a specific reason why it's not mmap'ed instead (java.nio style)? mmap is generally preferred when dealing with large files: it avoids the overhead of reading sector by sector, but is backed by the page cache (not a wasteful explicit copy).

mmap or not, Linux always has you work on a copy that is filled on-demand from the disk. This is the *page cache*. The least recently used pages are evicted 4KB at a time, and only under memory pressure, because the general Linux philosophy is "unused RAM is wasted RAM". In case of low memory, Linux auto-shrinks the cache based on the access patterns.

Unless this is deliberate due to an Android quirk?

**Comment by Iamlooker** [comment]
Can you test alpha build in the Telegram Group: https://t.me/droidifyc

**Comment by opk12** [comment]
The phone's owner agrees to a non-`CurrentVersion` update within F-droid. (I'm on Matrix, Telegram bridging is [trivial but opt-in](https://t2bot.io/telegram/).)

**Comment by opk12** [comment]
It's fixed in 0.5.9 Patch 1! Thank you.
